### PR TITLE
CVE-2016-4438

### PR DESCRIPTION
**Pull request by NateTheGr8Falk**
My answers for the CVE.

### CODE REVIEW

File: cves/CVE-2016-4438.yml [hunk 1]
```diff
@@ -45,7 +45,9 @@ description_instructions: |
   that outsiders to Struts would not understand. Technology like "regular
   expressions" is fine, and security phrases like "invalid write" are fine to
   keep too.
-description:
+description: |
+  An attacker can execute code in the application. It is possible by passing 
+  a malicious expression. If it didn't work a denial of service would ensue.
```
Comment (MattThyng) [inline cves/CVE-2016-4438.yml:50]: Some more detail in the description might be nice. How is this expression passed to struts? What problem in the code allows the remote code execution?

File: cves/CVE-2016-4438.yml [hunk 2]
```diff
@@ -144,7 +156,7 @@ interesting_commits:
     emerging themes?
     If there are no interesting commits, demonstrate that you completed this section
     by explaining what happened between the VCCs and the fix.
-  answer:
+  answer: No interesting commits found
```
Comment (MattThyng) [inline cves/CVE-2016-4438.yml:162]: Do you know what happened in the code between the VCC and the fix?

File: cves/CVE-2016-4438.yml [hunk 3]
```diff
@@ -207,4 +219,8 @@ mistakes:
     Use those questions to inspire your answer. Don't feel obligated to answer
     every one. Write a thoughtful entry here that those in the software
     engineering industry would find interesting.
-  answer:
+  answer: |
+    This is an input validation error. It is an input validation error because a user can
+    enter code to excute arbitraury code and failure with code can lead to a denial of service. This seems
+    to be a classic case of improper input validation because of the lack of input sanitization.
```
Comment (MattThyng) [inline cves/CVE-2016-4438.yml]: Did they have any input validation or sanitization beforehand or was it simply incomplete? Was this a design mistake or a coding mistake? A little more detail here might be nice :)

File: cves/CVE-2016-4438.yml (hunk 2 above)
Comment (NateTheGr8Falk) [reply cves/CVE-2016-4438.yml:162]: I looked and couldn't really find anything interesting. I  spent a while when for this I was doing research on this one.

File: cves/CVE-2016-4438.yml (hunk 3 above)
Comment (NateTheGr8Falk) [reply cves/CVE-2016-4438.yml]: I updated this on the last commit

File: cves/CVE-2016-4438.yml (hunk 1 above)
Comment (NateTheGr8Falk) [reply cves/CVE-2016-4438.yml:50]: I updated this on the last commit

Comment (jcotton42) [reply cves/CVE-2016-4438.yml:50]: Are you able to find an example of a malicious input? The unit tests may be enlightening.

**Comment by jcotton42** [comment]
I'd like to give this CVE 2 upvotes.

File: cves/CVE-2016-4438.yml (hunk 1 above)
Comment (NateTheGr8Falk) [reply cves/CVE-2016-4438.yml:50]: I searched but couldn't find any.

**Pull request merged**